src/mds/MDBalancer.cc

```diff
@@ -493,8 +493,14 @@ void MDBalancer::queue_split(const CDir *dir, bool fast)
 
     // Pass on to MDCache: note that the split might still not
     // happen if the checks in MDCache::can_fragment fail.
+    auto mds_bal_split_bits = g_conf->mds_bal_split_bits;
+    if (mds_bal_split_bits > 24) {
```

Comment:
Right, so this sanity check is already in `src/common/options.cc`. I don't see the need for this check.